### PR TITLE
Added DigraphMycielskian function

### PR DESCRIPTION
**Pull request by MTWhyte**
This pull request adds a function which takes a symmetric digraph, and returns its Mycielskian.

The Mycielskian of a symmetric digraph is a larger symmetric digraph constructed from it, which has greater chromatic number. For more information on the construction, see [here](https://en.wikipedia.org/wiki/Mycielskian).

### CODE REVIEW

File: doc/attr.xml [hunk 1]
```diff
@@ -1241,6 +1241,32 @@ gap> DigraphLoops(gr);
 </ManSection>
 <#/GAPDoc>
 
+<#GAPDoc Label="DigraphMycielskian">
+<ManSection>
+  <Attr Name="DigraphMycielskian" Arg="digraph"/>
+  <Returns>A digraph.</Returns>
+  <Description>
+    If <A>digraph</A> is a symmetric digraph, then <C>DigraphMycielskian</C> returns the Mycielskian of <A>digraph</A>. If <A>digraph</A> is not symmetric, then an error message is returned.<P/>
```
Comment (wilfwilson) [inline doc/attr.xml]: Could you please delete the second sentence here? We (seem to) have adopted the convention in this package that we don't include such sentences, describing than an error is given when the argument doesn't match the requirements.

File: doc/attr.xml [hunk 2]
```diff
@@ -1241,6 +1241,32 @@ gap> DigraphLoops(gr);
...
+    
+   The Mycielskian of a symmetric digraph is a larger symmetric digraph constructed from it, which has a larger chromatic number. For further information, see <URL> https://en.wikipedia.org/wiki/Mycielskian </URL>. <P/>
```
Comment (wilfwilson) [inline doc/attr.xml]: Could you please have the `<URL>` tags tight against the URL, i.e.
`<URL>https://en.wikipedia.org/wiki/Mycielskian</URL>`

File: doc/attr.xml [hunk 3]
```diff
@@ -1241,6 +1241,32 @@ gap> DigraphLoops(gr);
...
+
+    <Example><![CDATA[
+gap> D := Digraph([[2], [1]]);
```
Comment (wilfwilson) [inline doc/attr.xml]: Could you change this to `CycleDigraph(2)`; it's the same digraph, but perhaps makes it slightly easier for a reader to understand what graph it is 🙂

File: gap/attr.gi [hunk 4]
```diff
@@ -1336,6 +1336,42 @@ end);
 # It is the backend to IsBipartiteDigraph, Bicomponents, and DigraphColouring
 # for a 2-colouring
 
+InstallMethod(DigraphMycielskian, "for a symmetric digraph",
```
Comment (wilfwilson) [inline gap/attr.gi]: I think we really need a different version of this for mutable and immutable digraphs; the one for immutable digraphs would look like this:
```
if not IsSymmetricDigraph(D) or IsMultiDigraph(D) then
  (blah)
fi;
return MakeImmutable(DigraphMycielskian(DigraphMutableCopy(D)));
```
And the mutable version would be the version that exists at the moment.

As things stand at the moment, for an immutable digraph, the current function makes a new copy in memory of the existing digraph every time you add a vertex or an edge, which happens a large number of times and harms performance.

File: tst/standard/attr.tst [hunk 5]
```diff
@@ -1175,6 +1175,28 @@ gap> for i in [1 .. 50] do
...
+gap> D := DigraphMycielskian(D);
+<immutable digraph with 17 vertices, 112 edges>
+gap> ChromaticNumber(D);
+5
+gap> D1 := Digraph([[], [3], [2]]);
+<immutable digraph with 3 vertices, 2 edges>
+gap> D2 := Digraph([[], [3, 4], [2, 5], [2, 6], [3, 6], [4, 5, 7], [6]]);
+<immutable digraph with 7 vertices, 12 edges>
+gap> IsIsomorphicDigraph(DigraphMycielskian(D1), D2);
+true
```
Comment (wilfwilson) [inline tst/standard/attr.tst]: Could you also add a couple of tests of running this on a mutable digraph?

File: gap/attr.gi [hunk 6]
```diff
@@ -1336,6 +1336,42 @@ end);
...
+[IsDigraph],
+function(D)
+  local n, i, j;
+  if not IsSymmetricDigraph(D) or IsMultiDigraph(D) then
```
Comment (wilfwilson) [inline gap/attr.gi]: What does/should this do to digraphs that have loops?

Comment (MTWhyte) [reply gap/attr.gi]: That's a good point! If we take a graph with loops like `D := Digraph([[1, 2], [1]])`, then DigraphMycielskian(D) has a repeated edge, which I don't want. I'll fix this.

**Comment by wilfwilson** [comment]
Thanks for the changes @MTWhyte

**Pull request merged**